File: algorithms/python/rotateArray/rotateArray.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/rotate-array/
# Author: Miao Zhang
# Date:   2021-01-25

class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        nums[:] = nums[n - k:] + nums[: n - k] 

File: algorithms/cpp/rotateArray/rotateArray.cpp
// Source: https://leetcode.com/problems/rotate-array/
// Author: Miao Zhang
// Date:   2021-01-25

class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n = nums.size();
        k %= n;
        if (k == 0) return;
        reverse(nums.begin(), nums.end());
        reverse(nums.begin(), nums.begin() + k);
        reverse(nums.begin() + k, nums.end());
    }
};
